### ISSUE TITLE
Type unknown when instantiating type of TypeVar or Union

### ISSUE THREAD

**Issue by cdce8p** [microsoft/pylance-release#1422]
<h3> Environment data </h3>

-   Language Server version: 2021.6.1
-   OS and version: darwin x64
-   Python version: 3.10.0b2
-   python.analysis.indexing: undefined
-   python.analysis.typeCheckingMode: basic

<h3> Code Snippet </h3>

```py
from typing import TypeVar

class A: ...
class B: ...

T = TypeVar("T", A, B)

def func(cls: type[T]) -> T:
    instance1 = cls()
    reveal_type(instance1)  # Unknown
    return instance1


def func2(cls: type[A | B]) -> A | B:
    instance2 = cls()
    reveal_type(instance2)  # Unknown
    return instance2
```

<h3> Expected behaviour </h3>

```py
instance1: T
instance2: A | B
```

The mypy output
```
test.py:10: note: Revealed type is "test.A"
test.py:10: note: Revealed type is "test.B"
test.py:16: note: Revealed type is "Union[test.A, test.B]"
```

<h3> Actual behaviour </h3>

Both, `instance1` and `instance2`, are unknown.

<h3> Logs </h3>
--

**Comment by erictraut** [comment]
Thanks for the feature request. The type checker currently works with `type[A] | type[B]`, but I agree that it should also work with `type[A | B]` and `type[T]`.

**Comment by erictraut** [comment]
This will be addressed in the next release.

**Comment by bschnurr** [comment]
This issue has been fixed in version 2021.6.2, which we've just released. You can find the changelog here: https://github.com/microsoft/pylance-release/blob/main/CHANGELOG.md#202162-16-june-2021
